src/hooks/useContent.js

```diff
@@ -102,6 +107,42 @@ const useContent = ({
     }
   }, [loading])
 
+  /**
+   * update state value if different
+   * @param {any} state
+   * @param {any} newState
+   * @param {function} setState
+   */
+  function updateState(state, newState, setState) {
+    if (state !== newState) {
+      setState(newState)
+    }
+  }
+
+  useDeepCompareEffect(async () => {
+    // TRICKY: in the case of tWords there are two repos (tw for articles and twl for word list) and each one may have different branch
```

Comment:
Would it make sense to move this code to useUserBranch?

Reply:
Hmm...intriguing suggestion, looking into it.

Reply:
Only if you think it makes sense, otherwise I don't mind if you keep it here